app/models/collection.js

```diff
@@ -28,12 +26,27 @@ class Collection {
             _.extend(this, props);
         }
     }
+
+    deserializeBullets(bulletInstances) {
+        this.bullets = this.bullets.map(bulletId => {
```

Comment:
Not crazy about `this.bullets` being "hybrid". It seems like `this.bullets` should either be an array of ids or an array of bullet objects. Or possibly for easier id lookup, and object where the keys are bullet ids and the values are bullet objects.